src/python_testing/TC_CC_10_1.py

```diff
@@ -0,0 +1,484 @@
+            TestStep("9b", "TH sends a _RecallScene_ command to DUT with the _GroupID_ field set to _G~1~_, the _SceneID_ field set to 0x05 and the _TransitionTime_ omitted."),
+            TestStep("9c", "TH read _ColorLoopActive attribute_ from DUT."),
+            TestStep("9d", "TH read _ColorLoopDirection attribute_ from DUT."),
+            TestStep("9e", "TH read _ColorLoopTime attribute_ from DUT."),
+        ]
+        return steps
+
+    @async_test_body
+    async def test_TC_CC_10_1(self):
+        cluster = Clusters.Objects.ColorControl
+        attributes = cluster.Attributes
+
+        # Pre-Condition: Commissioning
+        self.step("0")
+
+        self.step("0a")
+
+        self.TH1 = self.default_controller
+        self.kGroupKeyset1 = 0x01a1
+        self.groupKey = Clusters.GroupKeyManagement.Structs.GroupKeySetStruct(
+            groupKeySetID=self.kGroupKeyset1,
+            groupKeySecurityPolicy=Clusters.GroupKeyManagement.Enums.GroupKeySecurityPolicyEnum.kTrustFirst,
+            epochKey0="0123456789abcdef".encode(),
+            epochStartTime0=1110000,
+            epochKey1="0123456789abcdef".encode(),
+            epochStartTime1=1110001,
+            epochKey2="0123456789abcdef".encode(),
+            epochStartTime2=1110002)
+
+        await self.TH1.SendCommand(self.dut_node_id, 0, Clusters.GroupKeyManagement.Commands.KeySetWrite(self.groupKey))
+
+        self.step("0b")
+        self.kGroup1 = 0x0001
+        mapping_structs: List[Clusters.GroupKeyManagement.Structs.GroupKeyMapStruct] = []
+
+        mapping_structs.append(Clusters.GroupKeyManagement.Structs.GroupKeyMapStruct(
+            groupId=self.kGroup1,
+            groupKeySetID=self.kGroupKeyset1,
+            fabricIndex=1))
+        result = await self.TH1.WriteAttribute(self.dut_node_id, [(0, Clusters.GroupKeyManagement.Attributes.GroupKeyMap(mapping_structs))])
+        asserts.assert_equal(result[0].Status, Status.Success, "GroupKeyMap write failed")
+
+        self.step("0c")
+        await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.Groups.Commands.RemoveAllGroups())
+
+        self.step("1a")
+        result = await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.Groups.Commands.AddGroup(self.kGroup1, "Group1"))
+        asserts.assert_equal(result.status, Status.Success, "Adding Group 1 failed")
+
+        self.step("1b")
+        result = await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.ScenesManagement.Commands.RemoveAllScenes(self.kGroup1))
+        asserts.assert_equal(result.status, Status.Success, "Remove All Scenes failed on status")
+        asserts.assert_equal(result.groupID, self.kGroup1, "Remove All Scenes failed on groupID")
+
+        self.step("1c")
+        result = await self.TH1.SendCommand(self.dut_node_id, 1, Clusters.ScenesManagement.Commands.GetSceneMembership(self.kGroup1))
```

Comment:
Not a problem, just an FYI - many of the tests you've written have the same setup. You can put multiple tests into a single test class if you want. Test classes have functions called "setup_test" and "teardown_test" that get run before and after each test. If you end up doing another one of these in python, that could be handy.

Reply:
Feel free to close this comment, btw, nothing to be done here just though it might be handy info.

Reply:
Noted thanks!